platform/pulpcore/app/viewsets/repository.py

```diff
@@ -51,6 +51,19 @@ def importers(self, request, name):
         serializer = ImporterSerializer(page, many=True, context={'request': request})
         return paginator.get_paginated_response(serializer.data)
 
+
+    def update(self, request, partial=True):
+        instance = self.get_object()
+        serializer = self.get_serializer(instance, data=request.data, partial=partial)
+        serializer.is_valid(raise_exception=True)
+        async_result = tasks.repository.update.apply_async_with_reservation(
+            tags.RESOURCE_REPOSITORY_TYPE, instance.name,
```

Comment:
Yes what we lock on needs to be immutable. Allowing people to change the repo name sounds valuable also. Together that would suggest we should lock on something other than the repo name. The one thing that I think is always immutable is the id field which is a uuid.

One thing to be aware of with ^ change is that users liked that we used the repo name so they could search for all outstanding tasks for a given repo. We need to allow that use case somehow.

This is a broad issue affecting all of our locks; we should look at all existing `apply_async_with_reservation` calls and change all of them. This is probably worth of a check-in thread on pulp-dev.